### PR TITLE
add copy metastore endpoint

### PR DESCRIPTION
**Pull request by nicflores**
Copy metastore endpoint added.

### CODE REVIEW

File: web/src/test/scala/quasar/api/services/MetastoreServiceSpec.scala [hunk 1]
```diff
@@ -77,6 +92,40 @@ class MetastoreServiceSpec extends quasar.Qspec {
       resp.as[String].unsafePerformSync must_=== s"Now using newly initialized metastore located at $expectedUrl"
       resp.status must_=== Ok
     }
+    "copy metastore" in {
+      // TODO: Use arbitraries
```
Comment (drostron) [inline web/src/test/scala/quasar/api/services/MetastoreServiceSpec.scala]: Still a relevant TODO?

File: web/src/test/scala/quasar/api/services/MetastoreServiceSpec.scala [hunk 2]
```diff
@@ -77,6 +92,40 @@ class MetastoreServiceSpec extends quasar.Qspec {
...
+
+      (for {
+        dstConn <- MetaStoreFixture.createNewTestMetaStoreConfig
+        dstMeta <- MetaStore.connect(dstConn, initializeOrUpdate = true, List(Schema.schema), Nil)
+                  .run.map(_.valueOr(e => scala.sys.error("Failed to initialize test metastore because: " + e.message)))
+        srcTrans = srcMeta.transactor
+        dstTrans = dstMeta.transactor
+        _     <- MetaStoreAccess.insertPathedMountConfig(pathedMountConfig).transact(srcTrans)
+        _     <- MetaStoreAccess.insertViewCache(pvc).transact(srcTrans)
+        req   <- Request(method = PUT, uri = Uri().+?("initialize").+?("copy")).withBody(dstConn.asJson)
```
Comment (drostron) [inline web/src/test/scala/quasar/api/services/MetastoreServiceSpec.scala]: We tend to omit `.` when using operator-like methods as in `Uri() +? "initialize" +? "copy"`

File: core/src/main/scala/quasar/metastore/Schema.scala [hunk 3]
```diff
@@ -25,6 +25,9 @@ import doobie.imports._
 import scalaz._, Scalaz._
 
 object Schema {
+  // NB: Changes to the schema should also be reflected in
+  // Metastore.copy, Queries, MetaStoreAccess, MetaStoreAccessSpec
```
Comment (drostron) [inline core/src/main/scala/quasar/metastore/Schema.scala]: `Queries` could be omitted as updates will naturally follow there

File: core/src/main/scala/quasar/metastore/MetaStore.scala [hunk 4]
```diff
@@ -16,25 +16,32 @@
 
 package quasar.metastore
 
+import argonaut.{DecodeJson, Json}
```
Comment (drostron) [inline core/src/main/scala/quasar/metastore/MetaStore.scala]: Import grouping

File: interface/src/test/scala/quasar/main/api/ApiSpec.scala [hunk 5]
```diff
@@ -26,13 +26,17 @@ import quasar.fs.PathError.pathNotFound
 import quasar.fs.mount._
 import quasar.fp.ski._
 import quasar.main._
+import quasar.metastore.MetaStore.{ShouldCopy, ShouldInitialize}
 import quasar.metastore.MetaStoreFixture
+
```
Comment (drostron) [inline interface/src/test/scala/quasar/main/api/ApiSpec.scala]: Import grouping

File: web/src/main/scala/quasar/api/services/metastore.scala [hunk 6]
```diff
@@ -21,11 +21,12 @@ import quasar.api._
 import quasar.db.DbConnectionConfig
 import quasar.fp.free._
 import quasar.main.{MainErrT, MetaStoreLocation}
-
 import org.http4s.dsl._
 import org.http4s.argonaut._
 import argonaut.Json
 import argonaut.Argonaut._
+import quasar.metastore.MetaStore.{ShouldCopy, ShouldInitialize}
```
Comment (drostron) [inline web/src/main/scala/quasar/api/services/metastore.scala]: Again

File: web/src/test/scala/quasar/api/services/MetastoreServiceSpec.scala [hunk 7]
```diff
@@ -21,22 +21,40 @@ import quasar.api._
...
+import quasar.fs.FileSystemType
+import quasar.fs.mount.{ConnectionUri, MountConfig, MountType}
+import quasar.fs.mount.cache.ViewCache
 import quasar.main._
-import quasar.metastore.{MetaStore, MetaStoreFixture, Schema}
+import quasar.metastore.{MetaStore, MetaStoreAccess, MetaStoreFixture, PathedMountConfig, PathedViewCache, Schema}
+import quasar.metastore.MetaStore.ShouldInitialize
+import quasar.sql._
+import quasar.Variables
+import java.time.Instant
```
Comment (drostron) [inline web/src/test/scala/quasar/api/services/MetastoreServiceSpec.scala]: Again

File: core/src/main/scala/quasar/metastore/MetaStore.scala [hunk 8]
```diff
@@ -20,21 +20,27 @@ import slamdata.Predef._
 import quasar.console.stdout
 import quasar.db._
 import quasar.fs.mount.MountingsConfig
-
-import argonaut._
 import doobie.util.transactor.Transactor
```
Comment (drostron) [inline core/src/main/scala/quasar/metastore/MetaStore.scala]: The `doobie` imports should be within the below group

File: web/src/test/scala/quasar/api/services/MetastoreServiceSpec.scala [hunk 9]
```diff
@@ -21,22 +21,37 @@ import quasar.api._
...
-  def service(persist: DbConnectionConfig => MainTask[Unit] = _ => ().point[MainTask]): Service[Request, Response] = {
-    val inter = Fixture.inMemFSWeb(persist = persist).unsafePerformSync
-    quasar.api.services.metastore.service[CoreEffIO].toHttpService(inter).orNotFound
-  }
+  def serviceWithMetaStore(persist: DbConnectionConfig => MainTask[Unit] = _ => ().point[MainTask]): (Service[Request, Response], MetaStore) =
+    (for {
+      ms    <- MetaStoreFixture.createNewTestMetastore()
+      inter <- Fixture.inMemFSWeb(persist = persist, metaRefT = TaskRef(ms))
+    } yield (quasar.api.services.metastore.service[CoreEffIO].toHttpService(inter).orNotFound, ms)).unsafePerformSync
+
```
Comment (nicflores) [inline web/src/test/scala/quasar/api/services/MetastoreServiceSpec.scala]: should we import `quasar.api.services.metastore`? maybe:
```
import quasar.api.services.metastore.{service => metastoreService}
```
or something like that.

**Pull request merged**